atlas-json-parent/atlas-json-core/src/main/java/io/atlasmap/json/inspect/SchemaInspector.java

```diff
@@ -0,0 +1,124 @@
+package io.atlasmap.json.inspect;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.atlasmap.json.v2.AtlasJsonModelFactory;
+import io.atlasmap.json.v2.JsonComplexType;
+import io.atlasmap.json.v2.JsonDocument;
+import io.atlasmap.json.v2.JsonFields;
+import io.atlasmap.v2.CollectionType;
+import io.atlasmap.v2.FieldStatus;
+import io.atlasmap.v2.FieldType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Iterator;
+import java.util.Map.Entry;
+
+/**
+ */
+public class SchemaInspector {
+
+    private JsonDocument jsonDocument = AtlasJsonModelFactory.createJsonDocument();
+    private static final Logger logger = LoggerFactory.getLogger(SchemaInspector.class);
+
+    public SchemaInspector() {
+    }
+
+    public JsonDocument getJsonDocument() {
+        return jsonDocument;
+    }
+
+    public void inspect(String schema) throws JsonInspectionException {
+        if (schema == null || schema.isEmpty()) {
+            throw new IllegalArgumentException("JSON schema cannot be null");
+        }
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode rootNode = objectMapper.readTree(schema);
+            
+            JsonNode header = rootNode.get("$schema");
+            if (header == null || !header.asText().startsWith("http://json-schema.org/")) {
+                throw new JsonInspectionException(String.format("The $schema property not found or invalid: '%s'", header.asText()));
+            }
+            JsonNode type = rootNode.get("type");
+            if (type == null || !type.asText().equals("object")) {
+                throw new JsonInspectionException(String.format("The property type '%s' is not supported for root node", type));
+            }
+
+            Iterator<Entry<String, JsonNode>> topFields = rootNode.get("properties").fields();
+            while (topFields.hasNext()) {
+                Entry<String, JsonNode> entry = topFields.next();
+                if (!entry.getValue().isObject()) {
+                    logger.warn("Ignoring non-object field '{}'", entry);
+                    continue;
+                }
+                logger.trace("--> Adding a field '{}' on a root node with value '{}'", entry.getKey(), entry.getValue());
+                JsonComplexType target = createJsonComplexType(entry.getKey(), entry.getValue(), null);
+                jsonDocument.getFields().getField().add(target);
+            }
+        } catch (Exception e) {
+            throw new JsonInspectionException(e);
+        }
+    }
+
+    private JsonComplexType createJsonComplexType(String name, JsonNode value, JsonComplexType parent) throws JsonInspectionException {
+        if (value.get("$ref") != null) {
```

Comment:
For results containing multiple objects (think search/query) it does use `$ref`, not sure if the Jackson JSON schema support might generate those on its own.